### ISSUE TITLE
Can't achieve widerface val score shown in the README when training from scratch.

### ISSUE THREAD

**Issue by waries** [ShiqiYu/libfacedetection.train#6]
I train this network using default parameters in train.py, then use final.pth to test on widerface val set scales=[1.], confidence_threshold=0.3. 
The scores I get are
Easy   Val AP: 0.7902144778486827
Medium Val AP: 0.7513014930849016
Hard   Val AP: 0.5267147634247762
Do you use some other tricks while training?

**Comment by xwu6614555** [comment]
def adjust_learning_rate_poly(optimizer, iteration, max_iter):
    """Sets the learning rate
    # Adapted from PyTorch Imagenet example:
    # https://github.com/pytorch/examples/blob/master/imagenet/main.py
    """
    lr = initial_lr * ( 1 - (iteration / max_iter)) * ( 1 - (iteration / max_iter))
    if ( lr < 1.0e-7 ):
      lr = 1.0e-7
    return lr

learning_rate is not adjusted at all!
add these codes before return lr：
    for param_group in optimizer.param_groups:
        param_group['lr'] = lr
Maybe this can solve the problem

**Comment by waries** [comment]
> 
> 
> def adjust_learning_rate_poly(optimizer, iteration, max_iter):
> """Sets the learning rate
> # Adapted from PyTorch Imagenet example:
> # https://github.com/pytorch/examples/blob/master/imagenet/main.py
> """
> lr = initial_lr * ( 1 - (iteration / max_iter)) * ( 1 - (iteration / max_iter))
> if ( lr < 1.0e-7 ):
> lr = 1.0e-7
> return lr
> 
> learning_rate is not adjusted at all!
> add these codes before return lr：
> for param_group in optimizer.param_groups:
> param_group['lr'] = lr
> Maybe this can solve the problem

Thanks for pointing that out, I'll try that and post the new result later.

**Comment by waries** [comment]
After modifying the lr code, the final result with scales=[1.], confidence_threshold=0.3 is
Easy   Val AP: 0.8285190221051939
Medium Val AP: 0.7929441126273071
Hard   Val AP: 0.5663795815749137

**Comment by ShiqiYu** [comment]
the learning rate is adapted according to epoches, not iterations

**Comment by ShiqiYu** [comment]
the learning rate is adapted according to epoches, not iterations

> After modifying the lr code, the final result with scales=[1.], confidence_threshold=0.3 is
> Easy Val AP: 0.8285190221051939
> Medium Val AP: 0.7929441126273071
> Hard Val AP: 0.5663795815749137

If you increase batch size to 256, and then fine-tune the mode, you will get a better result.

**Comment by waries** [comment]
> 
> 
> the learning rate is adapted according to epoches, not iterations
> 
> > After modifying the lr code, the final result with scales=[1.], confidence_threshold=0.3 is
> > Easy Val AP: 0.8285190221051939
> > Medium Val AP: 0.7929441126273071
> > Hard Val AP: 0.5663795815749137
> 
> If you increase batch size to 256, and then fine-tune the mode, you will get a better result.

Thank you for your guidance, I'll try it and post the new result. About fine-tune, does the initial lr need to be set to a smaller value?

**Comment by ShiqiYu** [comment]
You can use the same lr, but start from the 200th epoch.

**Comment by waries** [comment]
> 
> 
> You can use the same lr, but start from the 200th epoch.

Thank you so much for your guidance！
Here is my running code, is there anything I need to modify？
nohup python -u train.py -b 256 --num_workers 8 --gpu_ids 2 --lr 1e-4 --resume_net ./weights/yunet_200325_final.pth --resume_epoch 200 --max_epoch 500 --weight_filename_prefix weights/yunet_200326 > log200326.txt 2>&1 &

**Comment by waries** [comment]
> 
> 
> You can use the same lr, but start from the 200th epoch.

Thank you so much for your guidence！

**Comment by waries** [comment]
After fine-tune, the new result is
Easy   Val AP: 0.8321736396427939
Medium Val AP: 0.7988401596758341
Hard   Val AP: 0.579757106591874
I see professor Yu update the new result yesterday. So even though my new result is better than the original score which is tested on the model without fine-tune, there still exist a huge differences in the final result between the fine-tune model I trained and the official fine-tune model. 
I train the first edition model with batchsize 8 and 500 epochs, then use the final model to fine-tune with batch size 256 and start at epoch 200/500. I'm not sure which step was wrong，maybe the best way is to use pretrained model directly, LOL.

> 
> 
> You can use the same lr, but start from the 200th epoch.
